extensions/eda/plugins/event_source/file.py

```diff
@@ -15,69 +15,71 @@
 
 """
 
-import os
+import pathlib
 
 import yaml
 from watchdog.events import RegexMatchingEventHandler
 from watchdog.observers import Observer
 
 
-def send_facts(queue, filename: str) -> None:
+def send_facts(queue, filename: str) -> None: # noqa: ANN001
     """Send facts to the queue."""
-    with open(filename) as f:
+    with pathlib.Path(filename).open() as f:
         data = yaml.safe_load(f.read())
         if data is None:
             return
         if isinstance(data, dict):
             queue.put(data)
         else:
             if not isinstance(data, list):
-                msg = f"Unsupported facts type, expects a list of dicts found {type(data)}"
-                raise Exception(
-                    msg,
+                msg = (
+                    "Unsupported facts type, expects a list of dicts found "
+                    f"{type(data)}"
                 )
+                raise TypeError(msg)
             if not all(bool(isinstance(item, dict)) for item in data):
                 msg = f"Unsupported facts type, expects a list of dicts found {data}"
-                raise Exception(
-                    msg,
-                )
+                raise TypeError(msg)
             for item in data:
                 queue.put(item)
 
 
-def main(queue, args: dict) -> None:
+def main(queue, args: dict) -> None: # noqa: ANN001
     """Load facts from YAML files initially and when the file changes."""
-    files = [os.path.abspath(f) for f in args.get("files", [])]
+    files = [pathlib.Path(f).resolve().as_posix() for f in args.get("files", [])]
 
     if not files:
         return
 
     for filename in files:
         send_facts(queue, filename)
+    _observe_files(queue, files)
+
 
+def _observe_files(queue, files: list[str]) -> None: # noqa: ANN001
     class Handler(RegexMatchingEventHandler):
-        def __init__(self, **kwargs) -> None:
+        def __init__(self: "Handler", **kwargs) -> None: # noqa: ANN003
```

Comment:
Can we not use the same type as here https://github.com/ansible/event-driven-ansible/pull/135/commits/e7f4d98cc526840acd086542fb99a28d6a1ad178#diff-a614b4efc2dd1f32c82962f861ab13d16ac3a843d8b2c38a6136b21dcc3a8421R37?

Reply:
I believe this file is not compatible with ansible-rulebook engine. It should not be used. I put some simple fix to suppress the warning. This file should be removed or rewritten.